src/java/org/apache/cassandra/db/partitions/PartitionUpdate.java

```diff
@@ -240,6 +262,62 @@ static ByteBuffer toBytes(PartitionUpdate update, int version)
         }
     }
 
+    /**
+     *
+     * @return the estimated number of rows affected by this mutation
+     */
+    default int affectedRowCount()
+    {
+        // If there is a partition-level deletion, we intend to delete at least one row.
+        if (!partitionLevelDeletion().isLive())
+            return 1;
+
+        int count = 0;
+
+        // Each range delete should correspond to at least one intended row deletion.
+        if (deletionInfo().hasRanges())
+            count += deletionInfo().rangeCount();
+
+        count += rowCount();
+
+        if (!staticRow().isEmpty())
+            count++;
+
+        return count;
+    }
+
+    /**
+     *
+     * @return the estimated total number of columns that either have live data or are covered by a delete
+     */
+    default int affectedColumnCount()
+    {
+        // If there is a partition-level deletion, we intend to delete at least the columns of one row.
+        if (!partitionLevelDeletion().isLive())
+            return metadata().regularAndStaticColumns().size();
+
+        int count = 0;
+
+        // Each range delete should correspond to at least one intended row deletion, and with it, its regular columns.
+        if (deletionInfo().hasRanges())
+            count += deletionInfo().rangeCount() * metadata().regularColumns().size();
+
+        for (Row row : rows())
+        {
+            if (row.deletion().isLive())
+                // If the row is live, this will include simple tombstones as well as cells w/ actual data.
+                count += row.columnCount();
+            else
+                // We have a row deletion, so account for the columns that might be deleted.
+                count += metadata().regularColumns().size();
+        }
+
+        if (!staticRow().isEmpty())
+            count += staticRow().columnCount();
+
+        return count;
+    }
+
```

Comment:
For Review:
`affectedRowCount` and `affectedColumnCount` were both present in the 5.0 `PartitionUpdate` and here made default methods when `PartitionUpdate` became an interface.